rollup-operator/src/loop-manager.js

```diff
@@ -389,16 +416,98 @@ class LoopManager{
             this.timeouts.NEXT_STATE = 5000;
             // Check I am still the winner
             const deadlineReach = await this._blockDeadline(currentBlock);
-            if (deadlineReach){
+            const currentStateRoot = await this.rollupSynch.getCurrentStateRoot();
+            const currentOnchainHash = await this.rollupSynch.getMiningOnchainHash;
+
+            if (deadlineReach || BigInt(currentStateRoot) ==! this.infoCurrentBatch.batchData.getNewStateRoot() ||
+                BigInt(currentOnchainHash) ==! this.infoCurrentBatch.batchData.getOnChainHash())
+            {
                 // Cancel proof calculation
                 await this.cliServerProof.cancel();
                 this.state = state.SYNCHRONIZING;
                 this._resetInfoBatch();
+                this._resetInfoTx();
                 return;
             }
         }
     }
 
+    async _monitorTx(){
+
+        if (!this.overwriteTx)
+            return; 
+
+        this.overwriteTx = false;
+        if (this.currentTx.attempts > 5) {
+            this._errorTx("it has been overwritten more than 5 times");
+            return; 
+        }
+        const dataTx = await this.web3.eth.getTransaction(this.currentTx.txHash);
+        if (!dataTx) {
+            this._errorTx();
+            return;
+        }
+        if (dataTx.blocknumber) { //already mined!
+            this.timeouts.NEXT_STATE = 5000;
+            this.state = state.SYNCHRONIZING;
+            this._logTxOK();
+            this._resetInfoTx();
+            this._resetInfoBatch();
+            return;
+        }
+
+        const txSign = await this.opManager.signTransaction(this.currentTx.tx);
+        this._updateTx(txSign.transactionHash);
+        this._logResendTx();
+        const self = this;
+
+        try { //sanity check
+            await this.web3.eth.call(this.currentTx.tx);
+        } catch (error) { //error EVM transaction
+            if (error.message.includes("hash revealed not match current committed hash")) //already mined!
```

Comment:
- this error could be triggered due to blockchain re-organization or uncle hard fork
- I would handle it as an error

Reply:
i will check if the hashchain matches, if it do i will take it as correct if don't means that is are-organization, so i handle it as an error